test/uk/gov/hmrc/cdsreimbursementclaimfrontend/controllers/claims/SelectWhoIsMakingTheClaimControllerSpec.scala

```diff
@@ -105,201 +111,166 @@ class SelectWhoIsMakingTheClaimControllerSpec
             movementReferenceNumber = sampleEntryNumberAnswer()
           )
-        val (session, fillingOutClaim, _) = sessionWithClaimState(Some(answers))
-
-        val updatedJourney = fillingOutClaim.copy(draftClaim = draftC285Claim)
+        val (session, fillingOutClaim, _) = sessionWithClaimState(Some(answers), Some(numberOfClaims))
+        val updatedJourney                = fillingOutClaim.copy(draftClaim = draftC285Claim)
 
         inSequence {
           mockAuthWithNoRetrievals()
           mockGetSession(session.copy(journeyStatus = Some(updatedJourney)))
         }
 
         checkIsRedirect(
-          performAction(Seq("select-who-is-making-the-claim" -> "0")),
+          performAction(Seq("select-who-is-making-the-claim" -> "0"), journeyBindable),
           routes.EnterDetailsRegisteredWithCdsController
             .enterDetailsRegisteredWithCds()
         )
       }
 
-      "the user amends their answer" in {
-
-        def performAction(data: Seq[(String, String)]): Future[Result] =
-          controller.changeDeclarantTypeSubmit()(
-            FakeRequest().withFormUrlEncodedBody(data: _*)
-          )
-
-        val declarantType = DeclarantType.Importer
-
-        val answers: CompleteDeclarantTypeAnswer = CompleteDeclarantTypeAnswer(declarantType)
-
-        val draftC285Claim                = sessionWithClaimState(Some(answers))._3
+      "the user does not select an option" in forAll(testCases) { (numberOfClaims, journeyBindable) =>
+        val declarantType                 = DeclarantTypeAnswer.Importer
+        val answers                       = declarantType
+        val draftC285Claim                = sessionWithClaimState(Some(answers), Some(numberOfClaims))._3
           .copy(
             declarantTypeAnswer = Some(answers),
             movementReferenceNumber = sampleEntryNumberAnswer()
           )
-        val (session, fillingOutClaim, _) = sessionWithClaimState(Some(answers))
-
-        val updatedJourney = fillingOutClaim.copy(draftClaim = draftC285Claim)
+        val (session, fillingOutClaim, _) = sessionWithClaimState(Some(answers), Some(numberOfClaims))
+        val updatedJourney                = fillingOutClaim.copy(draftClaim = draftC285Claim)
 
         inSequence {
           mockAuthWithNoRetrievals()
           mockGetSession(session.copy(journeyStatus = Some(updatedJourney)))
         }
 
-        checkIsRedirect(
-          performAction(Seq("select-who-is-making-the-claim" -> "0")),
-          routes.CheckYourAnswersAndSubmitController
-            .checkAllAnswers()
+        checkPageIsDisplayed(
+          performAction(Seq.empty, journeyBindable),
+          messageFromMessageKey("select-who-is-making-the-claim.title"),
+          doc =>
+            doc
+              .select(".govuk-error-summary__list > li > a")
+              .text() shouldBe messageFromMessageKey(
+              s"select-who-is-making-the-claim.error.required"
```

Comment:
Can you define "select-who-is-making-the-claim" as key in the controller and reference this variable in the tests?

Reply:
done